static/src/javascripts/projects/common/modules/discussion/comment-count.js

```diff
@@ -68,10 +72,15 @@ define([
                 var format,
                     $node = bonzo(node),
                     url = $node.attr('data-discussion-url') || getContentUrl(node),
+                    shortUrl = $node.attr('data-loyalty-short-url').replace('/p/','') || '',
                     $container,
                     meta,
                     html;
 
+                if (shortUrl && ab.isInVariant('ParticipationHideHalfOfComments', 'hide-comments') &&  CommentBlocker.hideComments(shortUrl) ) {
```

Comment:
We shouldn't have checks outside of the ab test `canRun` function that change whether the ab test variant functionality is fired - ie: the check for `shortUrl` will modify whether we run the behaviour in the variant and will therefore skew the data so that users who are in the variant (because `canRun` has passed) will still be shown the comments.

You should move or duplicate the check for `shortUrl` into the `canRun` function so that **all** users in the variant will have hidden comments.